MAINTAINERS.md

```diff
@@ -77,8 +87,73 @@ Make frequent project releases to the community.
+
+> This is great work, Vacha! Would you be interested in becoming a [co-maintainer](https://github.com/opensearch-project/.github/blob/main/MAINTAINERS.md) on this project? I’d be honored to nominate you.
+
+Individuals accept the nomination by replying, or commenting, for example _"Thank you! I would love to."_
+
+### Public Confirmation
+
+Upon receiving 3 positive maintainer votes, and no vetoes, from other maintainers, and after having confirmed interest with the nominee, the maintainer opens a public issue in order to make the vote visible to the community. 
+
+> _Content from the above nomination._
+> 
+> Here is my +1.
+> 
+> If you support this nomination, please add your +1 or a comment. 
+> 
+> If you do not support this nomination, we invite you to e-mail all/some of the current [project maintainers](https://github.com/opensearch-project/OpenSearch/blob/main/MAINTAINERS.md) in the next 2 weeks. We will keep your feedback private to this group and discuss offline. We will not publicly post the reasons for a decision not to add this co-maintainer, or publicize your name. 
+> 
+> Anyone, including maintainers, can publicly add +1s as requested in the nomination issue if they support the nomination or contact the maintainers in private if they oppose. The issue stays open for 2 weeks.
+
+### Maintainer Decision
+
+If a positive decision has been reached, the nomination is accepted and the nominating maintainer can thus comment on the issue with _"maintainers has been added"_. If a decision cannot be reached for any reason, the nomination is rejected with _"a decision could not be reached"_ with no further explanation necessary.
+
+### Adding the New Maintainer 
+
+The Repo Admin adjusts the new maintainer’s permissions accordingly. The nominating maintainer opens a pull request to add the new maintainer to MAINTAINERS.md, and closes the nomination issue when that is merged and the individual has been given maintainer-level permissions.
+
+## Removing a Maintainer
+
+Removing a maintainer is a disruptive action that the community of maintainers should not undertake lightly. There are several reasons a maintainer might be removed from the project, such as inactivity, or taking actions that negatively impact the project.
+
+### Moving On
+
+There are plenty of reasons that might cause someone to want to take a step back or even a hiatus from a project. Existing maintainers can choose to leave the project at any time, with or without reason, by making a pull request to move themselves to the "Emeritus" section of MAINTAINERS.md, and asking an admin to remove their permissions.
+
+### Inactivity
```

Comment:
- What happens if there are no active maintainers? Is that okay? 

- Is there a limit to 'intends to reengage'? E.g. Lots of people have great intents but actions are different story, so if someone is inactive but 'intends to reengage' over and over does it just kick the can?

- The role of 'repo owner' is used here but not defined.

Reply:
> * What happens if there are no active maintainers? Is that okay?

That would make the project unmaintained, and it's on the admin to find additional maintainers. I am open to additional text for that if you want to suggest something.

> * Is there a limit to 'intends to reengage'? E.g. Lots of people have great intents but actions are different story, so if someone is inactive but 'intends to reengage' over and over does it just kick the can?

I guess so. Is that a problem?

> * The role of 'repo owner' is used here but not defined.

That's my bad. Replaced by repo admin. Thanks.

Reply:
How is an emeritus member reinstated? I see two options: 1) they are re-nominated by an existing maintainer, as though it's an entirely new addition. (Rationale: Merit doesn't expire, but it does get stale.) 2) They simply say "I want back in" and there's a nod from the rest of the team. The "nod" could be a vote, or it could be even more lightweight. FWIW, at Apache, an emeritus member returning requires a vote, but this is generally understood to be as lightweight as possible, and you don't say "no" without cause.

Reply:
Let's do a fast follow on emeritus, https://github.com/opensearch-project/.github/issues/61

Reply:
> That would make the project unmaintained, and it's on the admin to find additional maintainers. I am open to additional text for that if you want to suggest something.

Suggestion:

_If the repo is left without any maintainers, either by maintainer inactivity or moving on, the repo is considered unmaintained. The repo admin will seek out new maintainers and note the maintenance status in the repo README file._

Reply:
> > That would make the project unmaintained, and it's on the admin to find additional maintainers. I am open to additional text for that if you want to suggest something.
> 
> Suggestion:
> 
> _If the repo is left without any maintainers, either by maintainer inactivity or moving on, the repo is considered unmaintained. The repo admin will seek out new maintainers and note the maintenance status in the repo README file._

I like these words, but isn't it the admin responsibility and should be added to ADMINS.md?

Reply:
Thinking about it more, wdyt of https://github.com/opensearch-project/.github/pull/59/commits/8b3890381b75854efdbe5623b6f5e3348fdb24b6?

Reply:
@dblock 'If the inactive maintainer responds and intends to reengage, no action is required' -> If the inactive maintainer responds and reengages, no action is required

Reply:
Spot on, https://github.com/opensearch-project/.github/pull/59/commits/7fd6c112d5dac382bf60fadd1b359aeff465eea2.